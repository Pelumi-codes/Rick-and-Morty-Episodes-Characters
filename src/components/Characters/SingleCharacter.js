import { Image, Box, Text } from "@chakra-ui/react";
import { StyledFlex } from "../Styles";

export const SingleCharacterCard = ({ character, onClick }) => {
  return (
    <StyledFlex onClick={onClick} mr="2%" height="200px" mb={5} width="32%">
      <Image
        src={character?.image}
        alt="character image"
        borderRadius="5px 0 0 5px"
        flexBasis="40%"
        objectFit="contain"
        height="auto"
      />
      <Box
        flex="1"
        backgroundColor="rgb(59,62,67)"
        pl="10px"
        borderRadius="0 5px 5px 0"
      >
        <Text fontWeight={600} fontSize="18px" my="2%" color="white" mt="10%">
          Name: {character?.name}
        </Text>
        <Text  mt="4%" color="white">Status: {character?.status}</Text>
        <Text mt="4%" color="white">
          Species: {character?.species}
        </Text>
        <Text color="orange" mt="4%">Gender: {character?.gender}</Text>
      </Box>
    </StyledFlex>
  );
};
