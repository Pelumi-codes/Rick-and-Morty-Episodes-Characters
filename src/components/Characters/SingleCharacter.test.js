import React from "react";
import renderer from "react-test-renderer";

import  {SingleCharacterCard } from "./SingleCharacter";
it("renders list of characters", () => {
  const tree = renderer.create(<SingleCharacterCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
