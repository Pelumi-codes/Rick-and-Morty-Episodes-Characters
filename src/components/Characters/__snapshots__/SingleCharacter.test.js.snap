// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of characters 1`] = `
<div
  className="sc-bdvvtL ctcXYO css-1xlvfn4"
>
  <img
    alt="character image"
    className="chakra-image css-1g4aiwq"
  />
  <div
    className="css-1j42uzu"
  >
    <p
      className="chakra-text css-hp3kxj"
    >
      Name: 
    </p>
    <p
      className="chakra-text css-1a5tt7"
    >
      Status: 
    </p>
    <p
      className="chakra-text css-1a5tt7"
    >
      Species: 
    </p>
    <p
      className="chakra-text css-1lxxd5p"
    >
      Gender: 
    </p>
  </div>
</div>
`;
